### ISSUE TITLE
Barrier server doesn't start on startup in Linux (doesn't follow the Desktop Application Autostart Specification)

### ISSUE THREAD

**Issue by adam4235** [debauchee/barrier#1859]
### What happened?

I have barrier running as the server on Linux Mint XFCE edition 21.  I check the checkbox "Start barrier on startup", which I'm expecting means the server will start automatically when I reboot my computer.  But it didn't (verified by checking the log file).  I had to manually add it to the Application Autostart tab of the XFCE configuration.

I'm actually not sure if this is the meaning of the "Start barrier on startup" checkbox.  If it means something else - for example, if it means "start the barrierd background process when this interface dialog starts" - then that should be better clarified in the user interface.

### Version

v2.4.0

### If applicable, where did you install Barrier from?

Linux Mint package manager

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

Server is Linux Mint XFCE edition 21 (client is Windows 10, but that isn't relevant for this bug).

**Comment by adam4235** [comment]
I've been informed that there's a general specification for autostarting applications in Linux, which it seems Barrier isn't following.  So I updated the title, and here's a link to the specification:
https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html

**Comment by adam4235** [comment]
Note that I have Barrier installed via System Package in the Software Manager, not Flatpak.  According to the following XFCE bug I reported, this further confirms that Barrier is able to follow the spec by saving the .Desktop file in `$XDG_CONFIG_HOME/autostart`and `$XDG_CONFIG_DIRS/autostart`

https://gitlab.xfce.org/xfce/xfce4-session/-/issues/162

**Comment by yllekz** [comment]
A workaround for this is to add an entry to "Startup Applications" with command of ``barrier`` - though at startup it will display the menu UI and require you to click "start" to get it going (I don't know how to automate this, any ideas/help on that?)

**Comment by adam4235** [comment]
@yllekz Yeah I did mention that workaround in my above report ("I had to manually add it to the Application Autostart tab of the XFCE configuration.").  Regarding getting rid of the menu UI, it doesn't appear for me, maybe because I have the "Hide on Startup" and "Minimize to System Tray" checkboxes checked in the Barrier settings.

**Comment by yllekz** [comment]
@adam4235 where are those settings? The only settings panel I can locate is "advanced server settings" within "server configuration" and I do not see those options.

Edit: Scratch that, I found it. Was tucked away in the "Barrier" menu in the main window.

**Comment by UnixParadigm** [comment]
Bumping this to see if we can get this resolved, as this is as simple as barrier placing a .desktop file in the specified directory. Can confirm that enabling "Start Barrier at startup" does not function. Installed from flatpak on SteamOS.

**Comment by yllekz** [comment]
> Bumping this to see if we can get this resolved, as this is as simple as barrier placing a .desktop file in the specified directory. Can confirm that enabling "Start Barrier at startup" does not function. Installed from flatpak on SteamOS.

If you installed via flatpak there's a chance the app is significantly sandboxed and doesn't have the ability to do what it needs to do re: startup. On Linux Mint the startup feature works just fine for me when installing natively. If you install flatseal to check out permissions does the app have what it needs?

**Comment by adam4235** [comment]
> On Linux Mint the startup feature works just fine for me when installing natively.

Did this get fixed at some point then?  When you say it works, do you mean there's a .Desktop file getting added to your autostart directory automatically when you click the checkbox, and you didn't need to do anything manually in the OS?  I'm not using Linux Mint anymore so it's more difficult for me to check whether it's been fixed for me.

**Comment by UnixParadigm** [comment]
> > Bumping this to see if we can get this resolved, as this is as simple as barrier placing a .desktop file in the specified directory. Can confirm that enabling "Start Barrier at startup" does not function. Installed from flatpak on SteamOS.
> 
> If you installed via flatpak there's a chance the app is significantly sandboxed and doesn't have the ability to do what it needs to do re: startup. On Linux Mint the startup feature works just fine for me when installing natively. If you install flatseal to check out permissions does the app have what it needs?

![image](https://github.com/debauchee/barrier/assets/46477191/ba8c0fa0-b436-426f-a1b0-33c4476ad636)
I have given barrier permission to read write the autostart dir, even further more as testing I gave it full home read write.

![image](https://github.com/debauchee/barrier/assets/46477191/7cf69207-a63a-4a87-9775-b2b6b78dda87)
Here I have shown that the "Start Barrier on startup" option has been enabled, but the KDE autostart section does not show barrier (Does detect Warpinator, as it is present). Further more running `ls .config/autostart` shows it does only have Warpinator in there.

**Comment by UnixParadigm** [comment]
For those who stumble upon this thread in search of a solution, this is what I went with for those who use the flatpak version of Barrier.
You can simply copy the .desktop file from barriers flatpak application export directory to `.config/autostart` which AFAIK most sessions support. Single line command below to do so for you, run as standard user.
`cp "/var/lib/flatpak/app/com.github.debauchee.barrier/current/active/export/share/applications/com.github.debauchee.barrier.desktop" "${HOME}/.config/autostart"`
